torch/utils/data/dataloader.py

```diff
@@ -242,11 +269,23 @@ def _shutdown_workers(self):
         if not self.shutdown:
             self.shutdown = True
             self.done_event.set()
+            # if worker_manager_thread is waiting to put
+            if not self.data_queue.empty():
+                self.data_queue.get()
             for _ in self.workers:
                 self.index_queue.put(None)
+            # if all workers hang, no None is showed to worker_manager_thread,
+            # we put None to let worker_manager_thread exit
+            if self.worker_result_queue.empty():
+                self.worker_result_queue.put(None)
```

Comment:
Can't we just always put it to be sure? Having a non-empty queue doesn't imply that there is a worker that didn't hang - they might freeze after queueing some results